js/private/js_helpers.bzl

```diff
@@ -81,44 +81,15 @@ def gather_npm_linked_packages(srcs, deps):
         deps: dep targets; these typically come from the `deps` attribute of a rule
 
     Returns:
-        A `struct(direct, direct_files, transitive, transitive_files)` of direct and transitive npm linked packages & underlying files gathered
+        Depset of npm linked package files
     """
 
-    # npm_linked_packages
-    npm_linked_packages = [
+    return depset([], transitive = [
         target[JsInfo].npm_linked_packages
-        for target in srcs
-        if JsInfo in target and hasattr(target[JsInfo], "npm_linked_packages")
-    ]
-
-    # npm_linked_package_files
-    npm_linked_package_files = [
-        target[JsInfo].npm_linked_package_files
-        for target in srcs
-        if JsInfo in target and hasattr(target[JsInfo], "npm_linked_package_files")
-    ]
-
-    # transitive_npm_linked_packages
-    transitive_npm_linked_packages = depset([], transitive = npm_linked_packages + [
-        target[JsInfo].transitive_npm_linked_packages
         for target in srcs + deps
-        if JsInfo in target and hasattr(target[JsInfo], "transitive_npm_linked_packages")
-    ])
-
-    # transitive_npm_linked_package_files
-    transitive_npm_linked_package_files = depset([], transitive = npm_linked_package_files + [
-        target[JsInfo].transitive_npm_linked_package_files
-        for target in srcs + deps
-        if JsInfo in target and hasattr(target[JsInfo], "transitive_npm_linked_package_files")
+        if JsInfo in target and hasattr(target[JsInfo], "npm_linked_packages")
```

Comment:
Why is the `hasattr` needed? Shouldn't all `JsInfo` contain `.npm_linked_packages`?

Reply:
Starlark doesn't enforce that you set all fields when you create a provider so a custom rule in some 3rd party rule set could miss a field.